### ISSUE TITLE
Not compatible with latest Magisk

### ISSUE THREAD

**Issue by hexdefined** [skittles9823/QuickSwitch#73]
Magisk removed the `NVBASE` variable in https://github.com/topjohnwu/Magisk/commit/d33b077a13b34691a7f1fd3643336e09383ced78, causing `get_modpath.sh` in the app to fail, because it uses `util_functions.sh` and assumes `NVBASE` has been set.
```
get_modpath.sh[40]: [: 30: unexpected operator/operand
/modules_update/quickswitch
/modules/quickswitch
```

**Comment by osm0sis** [comment]
Can confirm this is the cause of most of the recent "Oops" error related regressions/issues. @skittles9823 @paphonb

Until the QuickSwitch ***app*** gets fixed users can add the following line to the bottom of /data/adb/magisk/util_functions.sh and it should get things working: 
```
NVBASE=/data/adb
```

**Comment by FERFAIL-TXZ** [comment]
> Can confirm this is the cause of most of the recent "Oops" error related regressions/issues. @skittles9823 @paphonb
> 
> Until the QuickSwitch _**app**_ gets fixed users can add the following line to the bottom of /data/adb/magisk/util_functions.sh and it should get things working:
> 
> ```
> NVBASE=/data/adb
> ```

This worked for me, thanks!

**Comment by Tr0ji** [comment]
OMG I haven't been able to use QuickSwitch for months now. Thanks heaps

**Comment by osm0sis** [comment]
> OMG I haven't been able to use QuickSwitch for months now. Thanks heaps

Yeah, not sure if everyone's just busy or what, but hopefully they recompile the APK with the fix included sometime soon...

CC: @skittles9823 @paphonb

**Comment by FabianMPunkt** [comment]
> Until the QuickSwitch _**app**_ gets fixed users can add the following line to the bottom of /data/adb/magisk/util_functions.sh and it should get things working:
> 
> ```
> NVBASE=/data/adb
> ```

Hey, please excuse my ignorance but where exactly is this file? 😅 I am not able to find it anywhere on LineageOS22.1. Do i need to build Magisk from source or something?

**Comment by osm0sis** [comment]
Magisk needs to be installed, and then it's exactly where I said.

**Comment by FabianMPunkt** [comment]
Sorry i had to use a different file manager that actually shows all root folders haha. I can change the recents provider now, but Lawnchair always crashes whenever i open the recents menu (Lawnchair 15.Dev #1922)

**Comment by Ajaymamtora** [comment]
> Magisk needs to be installed, and then it's exactly where I said.

Is there an alternative for KSU?

**Comment by LittleLogie** [comment]
> Can confirm this is the cause of most of the recent "Oops" error related regressions/issues. [@skittles9823](https://github.com/skittles9823) [@paphonb](https://github.com/paphonb)
> 
> Until the QuickSwitch _**app**_ gets fixed users can add the following line to the bottom of /data/adb/magisk/util_functions.sh and it should get things working:
> 
> ```
> NVBASE=/data/adb
> ```

I thought the issue was my rom, thanks it worked

**Comment by StarNumber12046** [comment]
> > Magisk needs to be installed, and then it's exactly where I said.
> 
> Is there an alternative for KSU?

I think the CLI, but it causes my Lawnchair to crash

**Comment by DethByte64** [comment]
> Can confirm this is the cause of most of the recent "Oops" error related regressions/issues. [@skittles9823](https://github.com/skittles9823) [@paphonb](https://github.com/paphonb)
> 
> Until the QuickSwitch _**app**_ gets fixed users can add the following line to the bottom of /data/adb/magisk/util_functions.sh and it should get things working:
> 
> ```
> NVBASE=/data/adb
> ```

This causes my lawnchair to crash.

Root manager: Magisk 28.1
Rom: ProtonAOSP 12.4.1
Launcher: Lawnchair Nightly 2179-b3e13e4

**Comment by StarNumber12046** [comment]
Yep

**Comment by clay-rip** [comment]
> Can confirm this is the cause of most of the recent "Oops" error related regressions/issues. [@skittles9823](https://github.com/skittles9823) [@paphonb](https://github.com/paphonb)
> 
> Until the QuickSwitch _**app**_ gets fixed users can add the following line to the bottom of /data/adb/magisk/util_functions.sh and it should get things working:
> 
> ```
> NVBASE=/data/adb
> ```

can confirm this is still working. crashes latest lawnchair 15 build, but downgrading to 14 beta 3 fixes this

**Comment by incogcyberpunk** [comment]
> Can confirm this is the cause of most of the recent "Oops" error related regressions/issues. [@skittles9823](https://github.com/skittles9823) [@paphonb](https://github.com/paphonb)
> 
> Until the QuickSwitch _**app**_ gets fixed users can add the following line to the bottom of /data/adb/magisk/util_functions.sh and it should get things working:
> 
> ```
> NVBASE=/data/adb
> ```

I can confirm, that this still , unbelievably is still the  fix for the `Error applying changes. Please send the logs to the developer.`, error encountered while setting even the latest `Lawnchair-15` as recents provider for Quickswitch.

BUT, this introuduces another issue of , `Lawnchair` not working properly and the recents not working with Quickswitch enabled by appling the above fix.
